File: sorting/py_sort/selection.py
def selection_sort(data):
    for i in range(len(data) - 1):
        min_index = i
        for j in range(i, len(data)):
            if data[j] < data[min_index]:
                min_index = j
        if min_index != i:
            data[min_index], data[i] = data[i], data[min_index]
    return data

File: sorting/c_plus/selection.cpp
#include "selection.h"
#include <iostream>

void selection_sort(std::vector<int>& arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        int min_index = i;
        for (int j = i + 1; j < arr.size(); j++)
        {
            if (arr[j] < arr[min_index])
            {
                min_index = j;
            }
        }
        if (min_index != i)
        {
            std::swap(arr[min_index], arr[i]);
        }
    }
}
